components/earnings/commons/earnings_base.lua

```diff
@@ -0,0 +1,192 @@
+-- @perYear - (optional) query all earnings per year and return the values in a lua table
+function Earnings.calculateForTeam(args)
+	args = args or {}
+	local team = args.team
+
+	if String.isEmpty(team) then
+		return 0
+	end
+	if not Logic.readBool(args.noRedirect) then
+		team = mw.ext.TeamLiquidIntegration.resolve_redirect(team)
+	end
+
+	local teamConditions = '([[participant::' .. team .. ']] OR [[extradata_participantteam::' .. team .. ']])'
+
+	return Earnings.calculate(teamConditions, args.year, args.mode, args.perYear, Earnings.divisionFactorTeam)
+end
+
+---
+-- Calculates earnings for this participant in a certain mode
+-- @participantCondition - the condition to find the player/team
+-- @year - (optional) the year to calculate earnings for
+-- @mode - (optional) the mode to calculate earnings for
+-- @perYear - (optional) query all earnings per year and return the values in a lua table
+-- @divisionFactor - divisionFactor function
+---
+-- customizable in case query has to be changed
+-- (e.g. SC2 due to not having a fixed number of players per team)
+function Earnings.calculate(conditions, year, mode, perYear, divisionFactor)
+	conditions = Earnings._buildConditions(conditions, year, mode)
+
+	if Logic.readBool(perYear) then
+		return Earnings.calculatePerYear(conditions, divisionFactor)
+	end
+
+	local lpdbQueryData = mw.ext.LiquipediaDB.lpdb('placement', {
+		conditions = conditions,
+		query = 'sum::prizemoney, mode',
+		groupby = 'mode asc'
+	})
+
+	local totalEarnings = 0
+
+	for _, item in ipairs(lpdbQueryData) do
+		if item['sum_prizemoney'] ~= nil then
+			local prizeMoney = item['sum_prizemoney']
+			totalEarnings = totalEarnings + (prizeMoney / divisionFactor(item['mode']))
+		end
+	end
+
+	return MathUtils._round(totalEarnings, 2)
+end
+
+---
+-- customizable in case query has to be changed
+-- (e.g. SC2 due to not having a fixed number of players per team)
+function Earnings.calculatePerYear(conditions, divisionFactor)
+	local totalEarningsOfYear = {}
+	local totalEarnings = {}
+	totalEarnings.total = 0
+
+	local offset = 0
+	local count = 5000
+	while count == 5000 do
+		local lpdbQueryData = mw.ext.LiquipediaDB.lpdb('placement', {
+			conditions = conditions,
+			query = 'prizemoney, mode, date',
+			limit = 5000,
+			offset = offset
+		})
+		for _, item in pairs(lpdbQueryData) do
+			local prizeMoney = tonumber(item.prizemoney) or 0
+			local year = string.sub(item.date, 1, 4)
+			prizeMoney = prizeMoney / divisionFactor(item['mode'])
+			totalEarningsOfYear[year] = (totalEarningsOfYear[year] or 0) + prizeMoney
+		end
+		count = #lpdbQueryData
+		offset = offset + 5000
+	end
+
+	for year, earningsOfYear in pairs(totalEarningsOfYear) do
+		totalEarnings[tonumber(year)] = MathUtils._round(earningsOfYear, 2)
+		totalEarnings.total = totalEarnings.total + earningsOfYear
+	end
+	totalEarnings.total = MathUtils._round(totalEarnings.total, 2)
+
+	return totalEarnings
```

Comment:
Copy from #742, I think we should consider returning two values instead of one.
`return sumEarnings, earningsByYear`